src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -116,6 +116,40 @@ protected async Task<IActionResult> ListDetectors(TResource resource, string que
             return Ok(listDetectorsTranslatedResponse);
         }
 
+        /// <summary>
+        /// Get Detector Meta Data with InternalOnly flag from Stoarge
+        /// </summary>
+        /// <param name="resource"></param>
+        /// <returns></returns>
+        protected async Task<IActionResult> ListDetectorsWithExtendMetaData(TResource resource)
+        {
+            DateTimeHelper.PrepareStartEndTimeWithTimeGrain(string.Empty, string.Empty, string.Empty, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage);
+            var cxt = PrepareContext(resource, startTimeUtc, endTimeUtc);
+            IEnumerable<ExtendedDefinition> detectors = new List<ExtendedDefinition>();
+            if (this.tableCacheService.IsStorageAsSourceEnabled())
```

Comment:
Will this IsStorageAsSourceEnabled() ever be false (e.g.. MoonCake)? If so, how should that be handled?

Reply:
If there is no storage account table, it will return an empty list.
Then we can in UI show as "unknown" or something else

Reply:
Or I can add the same logic as how we did in **ListDetectorsInternal()** to get internalOnly flag from SourceWatcher if storage account is disabled